lib/rpcenc/reader.go

```diff
@@ -55,62 +112,215 @@ func ReaderParamEncoder(addr string) jsonrpc.Option {
 		}
 		u.Path = path.Join(u.Path, reqID.String())
 
-		go func() {
-			// TODO: figure out errors here
+		rpcReader, redir := r.(*RpcReader)
+		if redir {
+			// if we have an rpc stream, redirect instead of proxying all the data
+			redir = rpcReader.redirect(u.String())
+		}
 
-			resp, err := http.Post(u.String(), "application/octet-stream", r)
-			if err != nil {
-				log.Errorf("sending reader param: %+v", err)
-				return
-			}
+		if !redir {
+			go func() {
+				// TODO: figure out errors here
+				for {
```

Comment:
We should probably add a bit of a backoff here, in order not to first a lot of requests in case they error out.